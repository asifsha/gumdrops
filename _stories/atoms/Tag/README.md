The `<Tag>` component is used to indicate active or selected items, filters or options. Refer to [this](http://design-prototypes.gumgum.com/black-tie/documentation/#icons-btl) page for icon names.

**Example**:

```jsx
<Tag
    context="primary"
    hasOption
    onOptionClick={() => {}}
    optionIcon="bt-times"
    size="sm"
    text="Sample Tag"
/>
```

| **prop name** | **description**                                                                                                         |
| ------------- | ----------------------------------------------------------------------------------------------------------------------- |
| className     | Class to add to the root element {string} {required}                                                                    |
| context       | Any of \`normal, primary, success, warning, danger\` or leave unset to get default appearance {string} {defaults to ''} |
| hasOption     | Indicate if the tag has an option button {boolean} {defaults to false}                                                  |
| onClick       | Click handler for the root element {function}                                                                           |
| onOptionClick | Click handler for the option button element {function}                                                                  |
| optionIcon    | Icon name for the option button {string} {defaults to 'bt-times'}                                                       |
| optionLabel   | Aria label for the option button {string} {defaults to 'Remove tag'}                                                    |
| size          | Any of \`sm, xs\` {string}                                                                                              |
| style         | Styles to add to the root element {object}                                                                              |
| text          | Tag display text {string}                                                                                               |
