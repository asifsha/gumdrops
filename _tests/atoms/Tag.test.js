/* globals mount */
import React from 'react';
import Tag from '../../components/atoms/Tag';

const defaultProps = {
    context: 'primary',
    className: 'custom-class',
    onClick: () => {},
    onOptionClick: () => {},
    hasOption: true,
    optionIcon: 'bt-times',
    optionLabel: 'Remove me',
    size: 'sm',
    style: { wdith: '100%' },
    text: 'Cool Tagz'
};

describe('Expect <Tag>', () => {
    it('to render', () => {
        const wrapper = mount(<Tag {...defaultProps} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('to render without option', () => {
        const props = {
            ...defaultProps,
            hasOption: false,
            onOptionClick: undefined,
            optionIcon: undefined
        };
        const wrapper = mount(<Tag {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
