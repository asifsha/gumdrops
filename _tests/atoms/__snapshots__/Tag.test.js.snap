// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Tag> to render 1`] = `
<Tag
  className="custom-class"
  context="primary"
  hasOption={true}
  onClick={[Function]}
  onOptionClick={[Function]}
  optionIcon="bt-times"
  optionLabel="Remove me"
  size="sm"
  style={
    Object {
      "wdith": "100%",
    }
  }
  text="Cool Tagz"
>
  <div
    className="gds-tag custom-class gds-tag--primary gds-tag--with-button gds-tag--with-button--sm gds-tag--sm"
    onClick={[Function]}
    style={
      Object {
        "wdith": "100%",
      }
    }
  >
    Cool Tagz
    <button
      aria-label="Remove me"
      className="gds-tag__option gds-tag__option--primary gds-tag__option--sm"
      onClick={[Function]}
    >
      <i
        className="btl bt-fw bt-times"
      />
    </button>
  </div>
</Tag>
`;

exports[`Expect <Tag> to render without option 1`] = `
<Tag
  className="custom-class"
  context="primary"
  hasOption={false}
  onClick={[Function]}
  optionIcon="bt-times"
  optionLabel="Remove me"
  size="sm"
  style={
    Object {
      "wdith": "100%",
    }
  }
  text="Cool Tagz"
>
  <div
    className="gds-tag custom-class gds-tag--primary gds-tag--sm"
    onClick={[Function]}
    style={
      Object {
        "wdith": "100%",
      }
    }
  >
    Cool Tagz
  </div>
</Tag>
`;
